### DISCUSSION TITLE
v2.7.0

### DISCUSSION THREAD

**Discussion by mienaiyami** [mienaiyami/yomikiru#63] [General]
## Downloads

[![Download setup](https://img.shields.io/badge/Windows%20Setup%20(exe)-Yomikiru--2.7.0--Setup.exe-brightgreen?logo=windows&logoColor=blue)](https://github.com/mienaiyami/yomikiru/releases/download/v2.7.0/Yomikiru-2.7.0-Setup.exe)

[![Download Portable](https://img.shields.io/badge/Windows%20Portable%20(zip)-Yomikiru--win32--2.7.0--Portable.zip-brightgreen?logo=windows&logoColor=blue)](https://github.com/mienaiyami/yomikiru/releases/download/v2.7.0/Yomikiru-win32-2.7.0-Portable.zip)

---
## Note

App name will be changing to "Yomikiru".

## Added

- Added canvas as image option to make stutter-free reader. Enable from settings.
- New reading mode "Right to Left".

## Changed

- Less stutter when scrolling in normal image mode.
- UI Enhancements.

## Fixed

- prev/next chapter not working when folder contain zip/cbz.
- Linking not working in theme maker.

---

#### Todo

- Add ability to change image quality in canvas mode for less RAM usage and performance.
- Write testing.
- Adjust app for debian.

---

**Full Changelog**: https://github.com/mienaiyami/yomikiru/compare/v2.6.0...v2.7.0

<hr /><em>This discussion was created from the release <a href='https://github.com/mienaiyami/yomikiru/releases/tag/v2.7.0'>v2.7.0</a>.</em>

**Comment by omodia** [comment]
> App name will be changing to "Yomikiru".
- I have changed the links and titles in reddit for the comments and post accordingly.
- What does 'Yomikiru' mean?

**Reply by mienaiyami** [reply]
Thanks.
It simply means "to read whole" or "to read through."

**Comment by omodia** [comment]
The DOMException is not being thrown now for the 800 width images, how did you do it? Not related to canvas, it is not checked.

**Reply by omodia** [reply]
I opened the same folder in the previous version and the DOMException is thrown.

**Reply by mienaiyami** [reply]
I tried decoding each image with some time gap. Previous one was also sequence but didnt work for some reason.

**Reply by omodia** [reply]
> I tried decoding each image with some time gap. Previous one was also sequence but didnt work for some reason.

### Stuttering issue is solved.
- I have tested the reader with 1200 width images and there is not stuttering of images at all and no DOMException is thrown.
- The DOMException is thrown for 2400w images (54 Images) one time only. But there is no stuttering at all. **The scrolling is very smooth.**
- To make sure, I have tested 116 images with 2400 width, the DOMException is thrown one time only and **there is no stuttering at all. The scrolling is also very smooth.**

**Reply by omodia** [reply]
**Thanks for solving the stuttering issue and your hard work.**
Since the stuttering issue is solved, will you still work on 'canvas' based image system? It is no longer necessary as the issue is solved.

**Reply by mienaiyami** [reply]
Theres lot more that can be done with canvas. Even now if you scroll too fast, reader still stutter but not with canvas.

**Comment by omodia** [comment]
> I tried decoding each image with some time gap. Previous one was also sequence but didnt work for some reason.

**A doubt?**
- If you increase the time gap, will the single DOMException also disappear?
- I would like to check it, if possible a mega link.

**Reply by mienaiyami** [reply]
No, I have tried and it's because of limit maybe. 
And one error mean that theres only one image that didnt decode, which is likely last one so there wont be any stutter.

**Reply by omodia** [reply]
I was just curious.

**Reply by mienaiyami** [reply]
And besides, if you increase time more either loading will take too much time or after loading it done it will still be decoding which might cause stuttering when scrolling.

**Reply by omodia** [reply]
Okay.

**Comment by omodia** [comment]
In the future, do you still plan to build the same app in 'tauri'?

**Reply by mienaiyami** [reply]
Very unlikely as of now, as over a quarter of the app depends on current framework, but cant say for far future.
If i were to start it in tauri, it might take month just to replace previous framework.
And after that comes matter of my own knowledge of the language tauri use, which is rust and I only know its basic compared to typescript which i know a lot more.

**Comment by omodia** [comment]
Do you want me to create a post in r/manga and r/mangapiracy?
Compared to r/manhua, r/manga have many numbers.
This time I won't delete the post because of downvotes.

**Reply by mienaiyami** [reply]
No, theres no need. Its not like im looking for more users and advertisement.
